signalwire-dotnet/Relay/Calling/Call.cs

```
@@ -165,7 +165,12 @@ public bool WaitFor(TimeSpan? timeout, params CallState[] states)
             API.API.Client.OnDisconnected += disconnectedCallback;
             OnStateChange += stateChangeCallback;
 
-            Task.Delay(timeout.Value, cancelDelay.Token).Wait();
+            try
+            {
+                Task.Delay(timeout.Value, cancelDelay.Token).Wait();
+            }
+            catch { }
```

I would probably explicitly catch `TaskCanceledException` here. It's not likely to swallow up any other meaningful exceptions, but the reader of the source should be alerted to what we expect.

---

It's deliberately swallowing everything because WaitFor should not throw any exceptions, it would confuse the end user if it failed in two different ways.  I suspect this is what is happening in another SDK that is confusing one of our own devs too as it's silently ignored up the stack on that SDK.

---

I'm going to tag in @bryanrite @edolix @lpradovera to get their thoughts on whether it should be a catch all that returns false, or if it should throw exceptions outside of the expected exception for cancelling the timeout.

---

I'm of split decision. I don't generally like rescuing everything because its hiding whats actually wrong. The issue with the other SDK is that there is _no_ feedback, nothing is raised, nothing happens. If I got a valid exception explaining the issue, I'm ok with that.

That all being said, this is on a `wait` command, so theres not much that can happen that an end-user can do anything about if there is a different exception, and presumably the next command, if any, would fail.

Either way is fine with me.